### PR TITLE
Use svg logo instead of png for better quality

### PR DESCRIPTION
**Pull request by Tyrrrz**
Currently, the logo looks very blurry because it's a resized PNG image:

image

SVG should look better when resized. Converted using a tool, so it's not very optimized.

Note: haven't tested it locally. Is there a way to deploy this on staging?

### CODE REVIEW

**Comment by bruno-garcia** [comment]
Thanks @Tyrrrz.. Since this is something the eyes can see, I'll defer to @joaopgrassi

**Comment by Tyrrrz** [comment]
Updated: running `ng serve` failed with:

```
An unhandled exception occurred: Invalid configuration object. Webpack has been initialised using a configuration object that does not match the API schema.
 - configuration.context: The provided value "f:\\Projects\\Softdev\\!Contrib\\nuget-trends\\src\\NuGetTrends.Web\\Portal" contains exclamation mark (!) which is not allowed because it's reserved for loader syntax.
   -> The base directory (absolute path!) for resolving the `entry` option. If `output.pathinfo` is set, the included pathinfo is shortened to this directory.
 - configuration.module.rules[12].include should be one of these:
   RegExp | string | function | [(recursive)] | object { and?, exclude?, include?, not?, or?, test? } | [RegExp | string | function | [(recursive)] | object { and?, exclude?, include?, not?, or?, test? }]
   -> One or multiple rule conditions
   Details:
    * configuration.module.rules[10].include[0]: The provided value "f:\\Projects\\Softdev\\!Contrib\\nuget-trends\\src\\NuGetTrends.Web\\Portal\\node_modules\\bulma\\css\\bulma.min.css" contains exclamation mark (!) which is not allowed because it's reserved for loader syntax.
    * configuration.module.rules[10].include[1]: The provided value "f:\\Projects\\Softdev\\!Contrib\\nuget-trends\\src\\NuGetTrends.Web\\Portal\\node_modules\\ngx-toastr\\toastr.css" contains exclamation mark (!) which is not allowed because it's reserved for loader syntax.
    * configuration.module.rules[10].include[2]: The provided value "f:\\Projects\\Softdev\\!Contrib\\nuget-trends\\src\\NuGetTrends.Web\\Portal\\node_modules\\font-awesome\\css\\font-awesome.min.css" contains exclamation mark (!) which is not allowed because it's reserved for loader syntax.
    * configuration.module.rules[10].include[3]: The provided value "f:\\Projects\\Softdev\\!Contrib\\nuget-trends\\src\\NuGetTrends.Web\\Portal\\src\\colors.scss" contains exclamation mark (!) which is not allowed because it's reserved for loader syntax.
    * configuration.module.rules[10].include[4]: The provided value "f:\\Projects\\Softdev\\!Contrib\\nuget-trends\\src\\NuGetTrends.Web\\Portal\\src\\styles.scss" contains exclamation mark (!) which is not allowed because it's reserved for loader syntax.
    * configuration.module.rules[10].include[0]: The provided value "f:\\Projects\\Softdev\\!Contrib\\nuget-trends\\src\\NuGetTrends.Web\\Portal\\node_modules\\bulma\\css\\bulma.min.css" contains exclamation mark (!) which is not allowed because it's reserved for loader syntax.
    * configuration.module.rules[10].include[1]: The provided value "f:\\Projects\\Softdev\\!Contrib\\nuget-trends\\src\\NuGetTrends.Web\\Portal\\node_modules\\ngx-toastr\\toastr.css" contains exclamation mark (!) which is not allowed because it's reserved for loader syntax.
    * configuration.module.rules[10].include[2]: The provided value "f:\\Projects\\Softdev\\!Contrib\\nuget-trends\\src\\NuGetTrends.Web\\Portal\\node_modules\\font-awesome\\css\\font-awesome.min.css" contains exclamation mark (!) which is not allowed because it's reserved for loader syntax.
    * configuration.module.rules[10].include[3]: The provided value "f:\\Projects\\Softdev\\!Contrib\\nuget-trends\\src\\NuGetTrends.Web\\Portal\\src\\colors.scss" contains exclamation mark (!) which is not allowed because it's reserved for loader syntax.
    * configuration.module.rules[10].include[4]: The provided value "f:\\Projects\\Softdev\\!Contrib\\nuget-trends\\src\\NuGetTrends.Web\\Portal\\src\\styles.scss" contains exclamation mark (!) which is not allowed because it's reserved for loader syntax.
    * configuration.module.rules[11].include[0]: The provided value "f:\\Projects\\Softdev\\!Contrib\\nuget-trends\\src\\NuGetTrends.Web\\Portal\\node_modules\\bulma\\css\\bulma.min.css" contains exclamation mark (!) which is not allowed because it's reserved for loader syntax.
    * configuration.module.rules[11].include[1]: The provided value "f:\\Projects\\Softdev\\!Contrib\\nuget-trends\\src\\NuGetTrends.Web\\Portal\\node_modules\\ngx-toastr\\toastr.css" contains exclamation mark (!) which is not allowed because it's reserved for loader syntax.
    * configuration.module.rules[11].include[2]: The provided value "f:\\Projects\\Softdev\\!Contrib\\nuget-trends\\src\\NuGetTrends.Web\\Portal\\node_modules\\font-awesome\\css\\font-awesome.min.css" contains exclamation mark (!) which is not allowed because it's reserved for loader syntax.
    * configuration.module.rules[11].include[3]: The provided value "f:\\Projects\\Softdev\\!Contrib\\nuget-trends\\src\\NuGetTrends.Web\\Portal\\src\\colors.scss" contains exclamation mark (!) which is not allowed because it's reserved for loader syntax.
    * configuration.module.rules[11].include[4]: The provided value "f:\\Projects\\Softdev\\!Contrib\\nuget-trends\\src\\NuGetTrends.Web\\Portal\\src\\styles.scss" contains exclamation mark (!) which is not allowed because it's reserved for loader syntax.
    * configuration.module.rules[11].include[0]: The provided value "f:\\Projects\\Softdev\\!Contrib\\nuget-trends\\src\\NuGetTrends.Web\\Portal\\node_modules\\bulma\\css\\bulma.min.css" contains exclamation mark (!) which is not allowed because it's reserved for loader syntax.
    * configuration.module.rules[11].include[1]: The provided value "f:\\Projects\\Softdev\\!Contrib\\nuget-trends\\src\\NuGetTrends.Web\\Portal\\node_modules\\ngx-toastr\\toastr.css" contains exclamation mark (!) which is not allowed because it's reserved for loader syntax.
    * configuration.module.rules[11].include[2]: The provided value "f:\\Projects\\Softdev\\!Contrib\\nuget-trends\\src\\NuGetTrends.Web\\Portal\\node_modules\\font-awesome\\css\\font-awesome.min.css" contains exclamation mark (!) which is not allowed because it's reserved for loader syntax.
    * configuration.module.rules[11].include[3]: The provided value "f:\\Projects\\Softdev\\!Contrib\\nuget-trends\\src\\NuGetTrends.Web\\Portal\\src\\colors.scss" contains exclamation mark (!) which is not allowed because it's reserved for loader syntax.
    * configuration.module.rules[11].include[4]: The provided value "f:\\Projects\\Softdev\\!Contrib\\nuget-trends\\src\\NuGetTrends.Web\\Portal\\src\\styles.scss" contains exclamation mark (!) which is not allowed because it's reserved for loader syntax.
    * configuration.module.rules[12].include[0]: The provided value "f:\\Projects\\Softdev\\!Contrib\\nuget-trends\\src\\NuGetTrends.Web\\Portal\\node_modules\\bulma\\css\\bulma.min.css" contains exclamation mark (!) which is not allowed because it's reserved for loader syntax.
    * configuration.module.rules[12].include[1]: The provided value "f:\\Projects\\Softdev\\!Contrib\\nuget-trends\\src\\NuGetTrends.Web\\Portal\\node_modules\\ngx-toastr\\toastr.css" contains exclamation mark (!) which is not allowed because it's reserved for loader syntax.
    * configuration.module.rules[12].include[2]: The provided value "f:\\Projects\\Softdev\\!Contrib\\nuget-trends\\src\\NuGetTrends.Web\\Portal\\node_modules\\font-awesome\\css\\font-awesome.min.css" contains exclamation mark (!) which is not allowed because it's reserved for loader syntax.
    * configuration.module.rules[12].include[3]: The provided value "f:\\Projects\\Softdev\\!Contrib\\nuget-trends\\src\\NuGetTrends.Web\\Portal\\src\\colors.scss" contains exclamation mark (!) which is not allowed because it's reserved for loader syntax.
    * configuration.module.rules[12].include[4]: The provided value "f:\\Projects\\Softdev\\!Contrib\\nuget-trends\\src\\NuGetTrends.Web\\Portal\\src\\styles.scss" contains exclamation mark (!) which is not allowed because it's reserved for loader syntax.
    * configuration.module.rules[12].include[0]: The provided value "f:\\Projects\\Softdev\\!Contrib\\nuget-trends\\src\\NuGetTrends.Web\\Portal\\node_modules\\bulma\\css\\bulma.min.css" contains exclamation mark (!) which is not allowed because it's reserved for loader syntax.
    * configuration.module.rules[12].include[1]: The provided value "f:\\Projects\\Softdev\\!Contrib\\nuget-trends\\src\\NuGetTrends.Web\\Portal\\node_modules\\ngx-toastr\\toastr.css" contains exclamation mark (!) which is not allowed because it's reserved for loader syntax.
    * configuration.module.rules[12].include[2]: The provided value "f:\\Projects\\Softdev\\!Contrib\\nuget-trends\\src\\NuGetTrends.Web\\Portal\\node_modules\\font-awesome\\css\\font-awesome.min.css" contains exclamation mark (!) which is not allowed because it's reserved for loader syntax.
    * configuration.module.rules[12].include[3]: The provided value "f:\\Projects\\Softdev\\!Contrib\\nuget-trends\\src\\NuGetTrends.Web\\Portal\\src\\colors.scss" contains exclamation mark (!) which is not allowed because it's reserved for loader syntax.
    * configuration.module.rules[12].include[4]: The provided value "f:\\Projects\\Softdev\\!Contrib\\nuget-trends\\src\\NuGetTrends.Web\\Portal\\src\\styles.scss" contains exclamation mark (!) which is not allowed because it's reserved for loader syntax.
 - configuration.output.path: The provided value "f:\\Projects\\Softdev\\!Contrib\\nuget-trends\\src\\NuGetTrends.Web\\Portal\\dist" contains exclamation mark (!) which is not allowed because it's reserved for loader syntax.
   -> The output directory as **absolute path** (required).
See "C:\Users\Tyrrr\AppData\Local\Temp\ng-8Hljr6\angular-errors.log" for further details.
```

**Comment by joaopgrassi** [comment]
Hum strange no idea what is that error. As the CI build works. Did you restore the pages using yarn or npm? We use yarn, so that might be it?

About the logo, thanks. The logo initially came from a freelancer I hired from Fiverr, but we had help of another contributor in the past that got us into the version that is today.

I looked locally and although it's better, there is some rough edges:
SVG:
image

Wheres the older is at least smooth (I agree it's blurry as hell..)
PNG
image

So not sure what is best. My OCD got triggered by the rough edges almost immediately :sweat_smile:

**Comment by Tyrrrz** [comment]
Actually, another option would probably be just to pre-resize the png file with a better resize algorithm.

For example this is the image resized to width=450, a lot less blurry than when it's resized at runtime

logo-inverted

**Comment by joaopgrassi** [comment]
Cool, if that improves it then I'm good with it :). If you want to push this new resized one then we can see how it looks.

**Comment by Tyrrrz** [comment]
Pushed an update

**Comment by joaopgrassi** [comment]
Ok I took a look, seems a bit better? Maybe my eye sight is just bad lol

image

File: src/NuGetTrends.Web/Portal/src/app/app.component.html [hunk 1]
```diff
@@ -4,7 +4,7 @@
       <div class="container">
         <div class="columns">
           <div class="column has-text-centered is-12">
-            <a [routerLink]="['/']"><img src="../assets/images/logo-inverted.png" width="400" alt="NuGet Trends brand logo"></a>
+            <a [routerLink]="['/']"><img src="../assets/images/logo-inverted-400.png" width="400" alt="NuGet Trends brand logo"></a>
```
Comment (joaopgrassi) [inline src/NuGetTrends.Web/Portal/src/app/app.component.html]: I guess now we can remove the width?

File: src/NuGetTrends.Web/Portal/src/app/core/footer/footer.component.html [hunk 2]
```diff
@@ -3,7 +3,7 @@
     <div class="columns">
       <div class="column has-text-centered">
         <a href="/">
-          <img src="../../assets/images/logo-inverted.png" width="350" alt="NuGet Trends brand logo">
+          <img src="../../assets/images/logo-inverted-350.png" width="350" alt="NuGet Trends brand logo">
```
Comment (joaopgrassi) [inline src/NuGetTrends.Web/Portal/src/app/core/footer/footer.component.html]: Here as well

**Review by joaopgrassi — CHANGES_REQUESTED** [review]

**Comment by Tyrrrz** [comment]
> Ok I took a look, seems a bit better? Maybe my eye sight is just bad lol
> 
> image

I don't know why, but in your screenshot neither of them look blurry. Here's how the current version looks for me (used Edge specifically to make sure it's not a browser issue):

image

You can see in my screenshot it's noticeably more blurry than in yours for some reason.

File: src/NuGetTrends.Web/Portal/src/app/app.component.html (hunk 1 above)
Comment (Tyrrrz) [inline src/NuGetTrends.Web/Portal/src/app/app.component.html]:
```suggestion
            <a [routerLink]="['/']"><img src="../assets/images/logo-inverted-400.png" alt="NuGet Trends brand logo"></a>
```

File: src/NuGetTrends.Web/Portal/src/app/core/footer/footer.component.html (hunk 2 above)
Comment (Tyrrrz) [inline src/NuGetTrends.Web/Portal/src/app/core/footer/footer.component.html]:
```suggestion
          <img src="../../assets/images/logo-inverted-350.png" alt="NuGet Trends brand logo">
```

**Review by joaopgrassi — APPROVED** [review]
Damn it does look awful on Chrome, my gosh! Sorry, I'm used to Firefox didn't even consider this case. It looks amazing now! Thanks a lot!

**Comment by bruno-garcia** [comment]
I'm glad the commits were not `wip`:

image

:)
Thanks for contributing @Tyrrrz

**Comment by joaopgrassi** [comment]
Oops..

**Comment by Tyrrrz** [comment]
> I'm glad the commits were not `wip`:

I promise to fix this next time ;)

**Comment by Tyrrrz** [comment]
> Damn it does look awful on Chrome, my gosh! Sorry, I'm used to Firefox didn't even consider this case. It looks amazing now! Thanks a lot!

Never considered Chrome to be so bad at resizing images :D

**Pull request merged**